crates/oxc_semantic/src/builder.rs

```diff
@@ -1671,6 +1671,8 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
         self.visit_formal_parameters(&func.params);
         if let Some(return_type) = &func.return_type {
             self.visit_ts_type_annotation(return_type);
+            // Resolve identifier references in return type.
+            self.resolve_references_for_current_scope();
```

Comment:
So far, we don't have extra logic in `Visitor` to resolve, so I think you should use the way https://github.com/oxc-project/oxc/pull/6388/ does

Reply:
I'm sorry. I forgot about #6388. I *think* it fixes the issue, I just needed to add some tests, and I didn't get around to doing that.